### PR TITLE
Added command line client for interacting with the Files API 

### PR DESCRIPTION
**Pull request by mjs**
This allows you to list, download and delete files using the Files API.

Also improved handling of 4xx errors.

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -143,3 +143,22 @@ def upload_recording(self, groupname, devicename, filename, props=None):
...
         self._check_response(r)
         return r.json()
+
+    def list_files(self):
+        url = urljoin(self._baseurl, "/api/v1/files")
+        r = requests.get(
+            url, params={"where": "{}", "order": '["id"]'}, headers=self._auth_header
+        )
+        self._check_response(r)
+        return r.json()["rows"]
```
Comment (gferraro) [inline api.py]: should be  ```return self._check_response(r)["rows"]```

File: files.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    return args
+
+
+def do_list(api):
+    for f in api.list_files():
+        print(format_file(f))
+
+
+def format_file(f):
```
Comment (gferraro) [inline files.py]: perhaps a different name like ```file_to_string``` might be more indicative of the function of this method

File: files.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    line = "{f[id]:3}[{f[type]}]: ".format(f=f)
```
Comment (gferraro) [inline files.py]: any reason for this not being an f-string?

File: api.py (hunk 1 above)
Comment (mjs) [reply api.py]: Good catch. Fixed.

File: files.py (hunk 2 above)
Comment (mjs) [reply files.py]: yep, this name isn't great. I went with `format_file_info`.

File: files.py (hunk 3 above)
Comment (mjs) [reply files.py]: not really. fixed.

**Pull request merged**